services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/UpdateItemOperation.java

```diff
@@ -90,7 +94,21 @@ public UpdateItemRequest generateRequest(TableSchema<T> tableSchema,
                                           r -> Optional.ofNullable(r.ignoreNulls()))
                                      .orElse(null);
 
-        Map<String, AttributeValue> itemMap = tableSchema.itemToMap(item, Boolean.TRUE.equals(ignoreNulls));
+        IgnoreNullsMode ignoreNullsMode = request.map(r -> Optional.ofNullable(r.ignoreNullsMode()),
+                                                     r -> Optional.ofNullable(r.ignoreNullsMode()))
+                                                        .orElse(IgnoreNullsMode.DEFAULT);
+
+        if (ignoreNullsMode == IgnoreNullsMode.SCALAR_ONLY) {
+            ignoreNulls = true;
+        }
+        Map<String, AttributeValue> itemMapImmutable = tableSchema.itemToMap(item, Boolean.TRUE.equals(ignoreNulls));
+        
+        // If ignoreNulls is set to true, check for nested params to be updated
+        // If needed, Transform itemMap for it to be able to handle them.
+
+        Map<String, AttributeValue> itemMap = ignoreNullsMode == IgnoreNullsMode.SCALAR_ONLY ?
```

Comment:
This line is different from the approved PR

Reply:
So just to confirm, MAPS_ONLY mode is what we originally supported with `ignoreNulls == true`, and `SCALAR_ONLY` is new support we added?

Reply:
Yes that is correct. with MAPS_ONLY, essentially a new map is created as part of a every item update regardless of whether its scalar or non-scalar. This does not work for scalar updates, and hence the SCALAR_ONLY